com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/topology.py

```diff
@@ -232,15 +235,45 @@ def parallel(self, width):
         
         Args:
             width (int): degree of parallelism
+            routing - denotes what type of tuple routing to use. 
+                ROUND_ROBIN: delivers tuples in round robin fashion to downstream operators
+                HASH_PARTIONED: delivers to downstream operators based on the hash of the tuples being sent
+                KEY_PARTIONED: delivers to downstream operators based on a hash provided by the user function provided
+            func - User defined function called when KEY_PARTIONED routing is specified.  The function provides an
```

Comment:
This is different to how the Java and SPL define parallel, where the key is defined, not the hash on a key.

Not saying this is wrong, but what's the reason for it being different. Really, the routing type is no longer KEY_PARTITIONED, but HASH_PARTITIONED with an application supplied hash.